examples/E2E-RAG-App/gradio_interface.py

```diff
@@ -0,0 +1,242 @@
+        providers = self.client.providers.list()
+        provider_id = providers["memory"][0].provider_id
+        collections = self.chroma_client.list_collections()
+
+        if any(col.name == self.memory_bank_id for col in collections):
+            print(f"The collection '{self.memory_bank_id}' exists.")
+        else:
+            print(
+                f"The collection '{self.memory_bank_id}' does not exist. Creating the collection..."
+            )
+            self.client.memory_banks.register(
+                memory_bank_id=self.memory_bank_id,
+                params={
+                    "embedding_model": "all-MiniLM-L6-v2",
+                    "chunk_size_in_tokens": 100,
+                    "overlap_size_in_tokens": 10,
+                },
+                provider_id=provider_id,
+            )
+            await self.load_documents()
+            print(f"Memory bank registered.")
+
+    async def load_documents(self):
+        """Load documents from the specified directory into memory bank."""
+        documents = []
+        for filename in os.listdir(self.docs_dir):
+            if filename.endswith((".txt", ".md")):
+                file_path = os.path.join(self.docs_dir, filename)
+                with open(file_path, "r", encoding="utf-8") as file:
+                    content = file.read()
+                    document = Document(
+                        document_id=filename,
+                        content=content,
+                        mime_type="text/plain",
+                        metadata={"filename": filename},
+                    )
+                    documents.append(document)
+
+        if documents:
+            self.client.memory.insert(
+                bank_id=self.memory_bank_id,
+                documents=documents,
+            )
+            print(f"Loaded {len(documents)} documents from {self.docs_dir}")
+
+    async def initialize_agent(self):
+        """Initialize the agent with model registration and configuration."""
+
+        if "1b" in MODEL_NAME:
+            model_name = "Llama3.2-1B-Instruct"
+        elif "3b" in MODEL_NAME:
+            model_name = "Llama3.2-3B-Instruct"
+        elif "8b" in MODEL_NAME:
+            model_name = "Llama3.1-8B-Instruct"
+        else:
+            model_name = MODEL_NAME
+
+        agent_config = AgentConfig(
+            model=model_name,
+            instructions="You are a helpful assistant that can answer questions based on provided documents. Return your answer short and concise, less than 50 words.",
+            sampling_params={"strategy": "greedy", "temperature": 1.0, "top_p": 0.9},
+            tools=[
+                {
+                    "type": "memory",
+                    "memory_bank_configs": [
+                        {"bank_id": self.memory_bank_id, "type": "vector"}
+                    ],
+                    "query_generator_config": {"type": "default", "sep": " "},
```

Comment:
is this parameter needed? is the default value OK? it is so confusing! (our bad, of course)